File: 20-valid-parentheses/20-valid-parentheses.java
import java.util.Stack;

class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();
    
        if (s.length() <=1) return false;
        
        for (char c : s.toCharArray()) {
            
            if (c == '(') st.push(')');
            
            else if (c == '[') st.push(']');
            
            else if (c == '{') st.push('}');
    
            else if (st.empty() || st.pop() != c) return false;
            
        }
        
        return st.empty();
    }
}

File: 20-valid-parentheses/20-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        
        for ch in s:
            if ch == '{': 
                stack.append('}')
            elif ch == '(': 
                stack.append(')')
            elif ch == '[':  
                stack.append(']')            
            else: 
                if len(stack) == 0 or ch != stack.pop():
                    return False
                
        return len(stack) == 0
                